src/test/rustdoc/sidebar-links-impls.rs

```diff
@@ -0,0 +1,49 @@
+// issue #56018: "Implementations on Foreign Types" sidebar items should link to specific impls
+
+#![crate_name = "foo"]
+
+// @has foo/trait.Foo.html
+// @has - '//*[@class="sidebar-title"]/a[@href="#foreign-impls"]' 'Implementations on Foreign Types'
+// @has - '//h2[@id="foreign-impls"]' 'Implementations on Foreign Types'
+// @has - '//*[@class="sidebar-links"]/a[@href="#impl-Foo-for-u32"]' 'u32'
+// @has - '//div[@id="impl-Foo-for-u32"]//h3[@class="code-header in-band"]' 'impl Foo for u32'
+// @has - '//*[@class="sidebar-links"]/a[@href="#impl-Foo-for-%26%27a%20str"]' "&'a str"
+// @has - '//div[@id="impl-Foo-for-%26%27a%20str"]//h3[@class="code-header in-band"]' "impl<'a> Foo for &'a str"
+pub trait Foo {}
+
+impl Foo for u32 {}
+
+impl<'a> Foo for &'a str {}
+
+// Issue #91118" "Implementors column on trait page is always empty"
+
+pub struct B;
+pub struct A;
+struct C;
+
+// @has foo/trait.Bar.html
+// @has - '//*[@class="sidebar-title"]/a[@href="#implementors"]' 'Implementors'
+// @has - '//h2[@id="implementors"]' 'Implementors'
+// @!has - '//*[@class="sidebar-title"]/a[@href="#foreign-impls"]'
+// @!has - '//h2[@id="foreign-impls"]'
+
+// @has - '//*[@class="sidebar-links"]/a[@href="#impl-Bar"]' '&A'
+// @has - '//*[@class="sidebar-links"]/a[@href="#impl-Bar-1"]' 'A'
+// @has - '//*[@class="sidebar-links"]/a[@href="#impl-Bar-2"]' 'B'
+
+// @has - '//div[@id="impl-Bar"]//h3[@class="code-header in-band"]' 'impl Bar for &A'
+// @has - '//div[@id="impl-Bar-1"]//h3[@class="code-header in-band"]' 'impl Bar for A'
+// @has - '//div[@id="impl-Bar-2"]//h3[@class="code-header in-band"]' 'impl Bar for B'
+pub trait Bar {}
+impl Bar for B {}
+impl Bar for A {}
+impl Bar for &A {}
+impl Bar for C {}
+
+// @has foo/trait.NotImpled.html
+// @!has - '//*[@class="sidebar-title"]/a[@href="#implementors"]'
+// FIXME: Is this the semantics we want
```

Comment:
Specificly the page is documented like

![image](https://user-images.githubusercontent.com/28781354/143664181-a3f41f4f-5b0a-4746-96bf-cbc883b7617a.png)


Because we only write the "Implementations on Foreign Types" header when they are present (https://github.com/rust-lang/rust/blob/ccce98535bd51687554be8a209b3a529ef84f7e1/src/librustdoc/html/render/print_item.rs#L755) wheras we don't for local impl (https://github.com/rust-lang/rust/blob/ccce98535bd51687554be8a209b3a529ef84f7e1/src/librustdoc/html/render/print_item.rs#L782-L790)

If theirs a reason for this, its fine to keep, otherwize I'll change it to be consistant

Reply:
It's because the `implementors` section also list implementations on foreign types (as you can see [here](https://doc.rust-lang.org/nightly/std/fmt/trait.Debug.html#impl-Debug-550) for example).